### PR TITLE
feat(docker-compose): update to v2

### PR DESCRIPTION
**Pull request by SeparateRecords**
~~This seems like a good opportunity for version based specs, will implement this before merge~~

### CODE REVIEW

File: src/docker-compose.ts [hunk 1]
```diff
@@ -15,40 +13,48 @@ const servicesGenerator: Fig.Generator = {
   splitOn: "\n",
 };
 
-const completionSpec: Fig.Spec = {
+const compose: Fig.Spec = {
   name: "docker-compose",
-  description: "Define and run multi-container applications with Docker",
+  description: "Docker Compose",
```
Comment (SeparateRecords) [inline src/docker-compose.ts:18]:
```suggestion
  description: "Define and run multi-container applications with Docker",
```

File: src/docker-compose.ts [hunk 2]
```diff
@@ -202,73 +280,79 @@ const completionSpec: Fig.Spec = {
...
-            isVariadic: true,
+            name: "user",
           },
         },
         {
-          name: ["-w", "--workdir"],
-          description: "DIR Path to workdir directory for this command",
+          name: ["--workdir", "-w"],
+          description: "Path to workdir directory for this command",
           args: {
```
Comment (SeparateRecords) [inline src/docker-compose.ts:338]: This should have `template: "folders"`

File: src/docker-compose.ts [hunk 3]
```diff
@@ -295,34 +382,81 @@ const completionSpec: Fig.Spec = {
...
+          },
         },
         {
           name: "--tail",
           description:
             "Number of lines to show from the end of the logs for each container",
           args: {
-            name: "Num of Lines",
-            suggestions: ["all"],
+            name: "tail",
```
Comment (SeparateRecords) [inline src/docker-compose.ts:415]:
```suggestion
            name: "lines",
```

File: src/docker-compose.ts [hunk 4]
```diff
@@ -295,34 +382,81 @@ const completionSpec: Fig.Spec = {
...
-          description: "Don't print prefix in logs",
+          name: ["--timestamps", "-t"],
+          description: "Show timestamps",
+        },
+        {
+          name: "--until",
+          description:
+            "Show logs before a timestamp (e.g. 2013-01-02T13:23:37Z) or relative (e.g. 42m for 42 minutes)",
+          args: {
+            name: "until",
```
Comment (SeparateRecords) [inline src/docker-compose.ts:428]:
```suggestion
            name: "timestamp",
```

File: src/docker-compose.ts [hunk 5]
```diff
@@ -331,38 +465,34 @@ const completionSpec: Fig.Spec = {
...
         { name: "private_port" },
       ],
       options: [
         {
-          name: "--protocol",
-          description: "Tcp or udp [default: tcp]",
+          name: "--index",
+          description:
+            "Index of the container if service has multiple replicas",
           args: {
```
Comment (SeparateRecords) [inline src/docker-compose.ts:485]: Should have `suggestions: ["tcp", "udp"]`

File: src/docker-compose.ts [hunk 6]
```diff
@@ -15,40 +13,48 @@ const servicesGenerator: Fig.Generator = {
   splitOn: "\n",
 };
 
-const completionSpec: Fig.Spec = {
+const compose: Fig.Spec = {
```
Comment (SeparateRecords) [inline src/docker-compose.ts:16]:
```suggestion
const completionSpec: Fig.Spec = {
```

File: src/docker-compose.ts [hunk 7]
```diff
@@ -1,5 +1,3 @@
-import { filepaths } from "@fig/autocomplete-generators";
-
 const servicesGenerator: Fig.Generator = {
```
Comment (SeparateRecords) [inline src/docker-compose.ts:1]: May want to update this?

**Comment by SeparateRecords** [comment]
@fedeci ready for review - decided not to go ahead with version based spec.

Reason is that with v2 there are two ways of running docker compose: `docker-compose` and `docker compose`
Version based specs can't determine which to load based on the command being run, so the "solution" is a really ugly hack: have two version-based specs, one for docker-compose directly and the other loaded by `docker`

**Comment by fedeci** [comment]
> Reason is that with v2 there are two ways of running docker compose: `docker-compose` and `docker compose` Version based specs can't determine which to load based on the command being run, so the "solution" is a really ugly hack: have two version-based specs, one for docker-compose directly and the other loaded by `docker`

Isn't `loadSpec` properly working?

**Pull request merged**